src/main/java/org/digma/intellij/plugin/toolwindow/recentactivity/RecentActivityService.java

```diff
@@ -1,47 +1,389 @@
 
+    private final Logger logger = Logger.getInstance(RecentActivityService.class);
+    private final Icon icon = AppIcons.TOOL_WINDOW_OBSERVABILITY;
+    private final Icon iconWithGreenDot = ExecutionUtil.getLiveIndicator(icon);
+    private final String localHostname;
     private final Project project;
-    private JBCefBrowser jbCefBrowser;
+    private final AnalyticsService analyticsService;
 
     //the recent activity code is not managed in one place that is accessible from the plugin code
     // like a project service, so currently need an init task.
     // it is used to send live data in case the live view button was clicked before the recent activity window was initialized.
-    private MyInitTask initTask = null;
+    private MyInitTask initTask;
+    private Timer myLiveDataTimer;
+    private Timer activityFetchingTimer;
+    private JBCefBrowser jbCefBrowser;
+    private RecentActivityResult latestActivityResult;
+    private boolean webAppInitialized;
 
-    private Timer myLiveDataTimer = null;
+    public static RecentActivityService getInstance(Project project) {
+        return project.getService(RecentActivityService.class);
+    }
 
     public RecentActivityService(Project project) {
         this.project = project;
+        //initialize AnalyticsService early so the UI already can detect the connection status when created
+        this.analyticsService = project.getService(AnalyticsService.class);
+        this.localHostname = CommonUtils.getLocalHostname();
+        this.latestActivityResult = new RecentActivityResult(null, new ArrayList<>());
     }
 
+    public void startFetchingActivities() {
+        if (activityFetchingTimer != null)
+            return;
 
-    public static RecentActivityService getInstance(Project project){
-        return project.getService(RecentActivityService.class);
+        var activityFetchingTask = new TimerTask() {
+            @Override
+            public void run() {
+                fetchRecentActivities();
+            }
+        };
+        activityFetchingTimer = new Timer();
+        activityFetchingTimer.scheduleAtFixedRate(activityFetchingTask, 0, FETCHING_LOOP_INTERVAL);
+    }
+
+    public Content createTabContent() {
+        if (!JBCefApp.isSupported()) {
+            // Fallback to an alternative browser-less solution
+            return null;
+        }
+
+        var contentFactory = ContentFactory.getInstance();
+        var editorService = project.getService(EditorService.class);
+        var customViewerWindow = new CustomViewerWindow(project, RESOURCE_FOLDER_NAME, new HashMap<>() {{
+            put(RECENT_EXPIRATION_LIMIT_VARIABLE, RECENT_EXPIRATION_LIMIT_MILLIS);
+        }});
+        jbCefBrowser = customViewerWindow.getWebView();
+
+        JBCefClient jbCefClient = jbCefBrowser.getJBCefClient();
+
+        jbCefBrowser.getCefBrowser().setFocus(true);
+
+        RecentActivityService.getInstance(project).setJcefBrowser(jbCefBrowser);
+
```

Comment:
This is redundant call , jbCefBrowser is a member of this class. and setJcefBrowser method can be deleted

Reply:
missed it, thanks